frontend/javascripts/oxalis/model/reducers/skeletontracing_reducer_helpers.js

```diff
@@ -69,6 +69,17 @@ function getMaximumTreeId(trees: TreeMap): number {
   return _.max(_.map(trees, "treeId"));
 }
 
+function getNextTreeWithHigherId(treeId: number, trees: TreeMap): ?Tree {
+  let foundTree: ?Tree = null;
+  const upperLimit = getMaximumTreeId(trees);
+  for (let currentTreeId = treeId + 1; !foundTree && currentTreeId <= upperLimit; ++currentTreeId) {
+    if (trees[currentTreeId] != null) {
+      foundTree = trees[currentTreeId];
+    }
+  }
+  return foundTree;
```

Comment:
An alternative suggestion for this method (because treeIds could be very sparse in theory):

```javascript
const sortedTreeIds = Object.keys(trees).sort();
if (sortedTreeIds.length === 0) return 0;
// Uses a binary search to determine the lowest index at which treeId should be inserted into sortedTreeIds in order to maintain its sort order.
// This corresponds to the original index of the deleted treeId.
const originalIndex = _.sortedIndex(sortedTreeIds, treeId);
const higherOrNearestId = Math.min(originalIndex, sortedTreeIds.length - 1);
return sortedTreeIds[higherOrNearestId];
```

Your call, both methods are fine with me.

Reply:
Thanks a lot for your suggestion :+1: